Reminder-App/Reminder-App/ReminderListViewController.swift

```diff
@@ -9,32 +9,217 @@ import UIKit
 
 class ReminderListViewController: UITableViewController {
     
+    // MARK: Outlets
+    @IBOutlet var filterSegmentedControl: UISegmentedControl!
+    @IBOutlet var progressContainerView: UIView!
+    @IBOutlet var percentIncompleteView: UIView!
+    @IBOutlet var percentCompleteView: UIView!
+    @IBOutlet var percentCompleteHeightConstraint: NSLayoutConstraint!
+    
+    // MARK: Actions
+    @IBAction func addButtonTriggered(_ sender: UIBarButtonItem) {
+        addReminder()
+    }
+    
+    @IBAction func segmentControlChanged(_ sender: UISegmentedControl) {
+        reminderListDataSource?.filter = filter
+        
+        tableView.reloadData()
+        self.refreshProgressView()
+        refreshBackground()
+    }
+    
     // MARK: Private Properties
     private var reminderListDataSource: ReminderListDataSource?
+    private var filter: ReminderListDataSource.Filter {
+        return ReminderListDataSource.Filter(rawValue: filterSegmentedControl.selectedSegmentIndex) ?? .today
+    }
     
-    // MARK: Static Properties
+    // MARK: Public Properties
     static let showDetailSegueIdentifier = "ShowReminderDetailSegue"
     static let reminderListCellIdentifier = "ReminderListCell"
```

Comment:
same issue of using CellIdentifier string rather than using Extended Computed Property of UITableViewCell